### PR TITLE
Allow a proxy.connector to be injected into the N…

### PR DESCRIPTION
**Pull request by krismcqueen**
…ioSocketConnector to handle cases were a proxy is configured in the JVM to handle certain connections.  See java.net.Proxy and java.net.ProxySelector.Also fix up some minor formatting.

### CODE REVIEW

**Comment by krismcqueen** [comment]
@jitsni the CE part of the fix to enable proxy transport in EE...

File: transport/nio/src/main/java/org/kaazing/gateway/transport/nio/NioSocketConnector.java [hunk 1]
```diff
@@ -97,6 +104,12 @@ protected BridgeServiceFactory initBridgeServiceFactory() {
             ResourceAddress address, IoHandler handler,
             IoSessionInitializer<T> initializer) {
 
+        if (proxyConnector != null) {
+            ConnectFuture future = proxyConnector.connect(address, handler, initializer);
+            if (future != null) {
```
Comment (jitsni) [inline transport/nio/src/main/java/org/kaazing/gateway/transport/nio/NioSocketConnector.java]: I see that you moved the code to create proxy:// to proxy transport component. That seems to be ok but it means the Connector#connect is returning a null ConnectFuture. I don't think any Connector.connect returns null future. We don't have javadoc on the connect methods to explain the contract but my guess is that it returns non-null ConnectFuture object. This is something to think about it.

Other option is to create proxy address here.

**Comment by jitsni** [comment]
Overall looks good.

**Comment by brennangaunce** [comment]
Do we have a test that validates this feature?  Please include positive and negative tests with this sprint.

**Comment by jitsni** [comment]
The ProxySelectorIT test in EE will verify this. In CE, this is just an extension point.

**Pull request merged**